p2p/swarm.go

```diff
@@ -178,7 +180,7 @@ func newSwarm(ctx context.Context, config config.Config, newNode bool, persist b
 
 	s.cPool = cpool
 
-	s.gossip = gossip.NewProtocol(config.SwarmConfig, s, s.LocalNode().PublicKey(), s.lNode.Log)
+	s.gossip = gossip.NewProtocol(config.SwarmConfig, s.config.BufferSize, s, s.LocalNode().PublicKey(), s.lNode.Log)
```

Comment:
I find it kinda troubling that we use this BufferSize for many things, we need to revisit that to make sure that its ok or create a new var.

Reply:
this buffer size is used for the buffer of each protocol messages queue. while priority q gets messages from all protocols.. we probably need to rethink this

Reply:
1. Yes, we should think also if buffer size should be put in SwarmConfig instead.
2. PQ uses this buffer size for each queue. It is true that each queue may have more than one protocol on it.

Can you open an issue summarizing your concerns with actions? @y0sher

Reply:
BufferSize is a param of the p2p config. it shouldn't be in `SwarmConfig` most of `SwarmConfig` params are related to discovery hence should be renamed to `DiscoveryConfig`. this will happen later before we release testnet, there should be an issue for that.
The concern I was raising is that BufferSize might change because it's needed for something else. also passing `BufferSize` to `gossip.Protocol` looks like it has something to do with the gossip buffer size and doesn't indicate anything about it used only for the prioq.

I prefer that you use the default value used for the propagation Q, or double it (which will eventually pass values to the prioQ). checkout `propagateHandleBufferSize` 

I don't see the point in adding new constructor argument to `Protocol` this way.

Reply:
I see that the propagation queue is limited to propagateHandleBufferSize. I can use it as the size of each queue instead of buffer size from the config. This way the PQ is ensured to have enough buffer (even in each sub q) to hold what propQ supports. Is that OK? (no param of course, I'll use this constant instead.  @y0sher

Reply:
yes you're right. no param. though when and if we parametrize they will be the same.

Reply:
Ok